test/refda/test_exec.c

```diff
@@ -406,3 +406,62 @@ void test_refda_exec_wait_cond(int delay_ms)
 
     cace_ari_deinit(&target);
 }
+
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:/TD/1, ari:/TD/60
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "820D01", "820D183C", 1, true)
+void test_refda_exec_time_based_rule(const char *actionhex, const char *starthex, const char *periodhex,
+                                     int max_exec_count, bool init_enabled)
+{
+    refda_amm_tbr_desc_t tbr;
+    {
+        struct timespec nowtime;
+        clock_gettime(CLOCK_REALTIME, &nowtime);
+
+        refda_amm_tbr_desc_init(&tbr);
+        TEST_ASSERT_EQUAL_INT(0, test_util_ari_decode(&(tbr.action), actionhex));
+        TEST_ASSERT_EQUAL_INT(0, test_util_ari_decode(&(tbr.start_time), starthex));
+        TEST_ASSERT_EQUAL_INT(0, test_util_ari_decode(&(tbr.period), periodhex));
+        tbr.max_exec_count      = max_exec_count;
+        tbr.absolute_start_time = nowtime;
+
+        refda_exec_tbr_enable(&agent, &tbr);
+    }
+
+    refda_exec_worker_iteration(&agent);
+    refda_exec_waiting(&agent); // run cleanup
+    refda_amm_tbr_desc_deinit(&tbr);
+}
+*/
+
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:undefined, ari:/TD/1
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "F7", "820D01", 1, true, 1)
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:/AC/(/BOOL/true), ari:/TD/1
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "8211818201F5", "820D01", 1, true, 0)
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:/AC/(/INT/1), ari:/TD/1
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "821181820401", "820D01", 1, true, 0)
+// ari:/AC/(//65535/10/CTRL/1,//65535/10/CTRL/2), ari:/AC/(/BOOL/false), ari:/TD/1
+TEST_CASE("8211828419FFFF0A22018419FFFF0A2202", "8211818201F4", "820D01", 1, true, 0)
+void test_refda_exec_state_based_rule(const char *actionhex, const char *condhex, const char *min_interval_hex,
```

Comment:
It's more complex, but can a test function be added that evaluates an EDD (defined in the test fixture) so that we can see one rule evaluation that does not execute the action, change the EDD state, and a second eval that does exec?

Reply:
More complex but definitely a good test. Alright, a new test has been added along those lines.